starknet.go

```diff
@@ -0,0 +1,235 @@
+		Program           string            `json:"program"`
+	} `json:"contract_definition"`
+}
+
+type EntryPointsByType struct {
+	Constructor []struct {
+		Offset   string `json:"offset"`
+		Selector string `json:"selector"`
+	} `json:"CONSTRUCTOR"`
+	External []struct {
+		Offset   string `json:"offset"`
+		Selector string `json:"selector"`
+	} `json:"EXTERNAL"`
+	L1Handler []interface{} `json:"L1_HANDLER"`
+}
+
+/*
+	'call_contract' wrapper and can accept a blockId in the hash or height format
+*/
+func (sg *StarknetGateway) Call(ctx context.Context, tx Transaction, opts *BlockOptions) ([]string, error) {
+	tx.EntryPointSelector = BigToHex(GetSelectorFromName(tx.EntryPointSelector))
+	if len(tx.Calldata) == 0 {
+		tx.Calldata = []string{}
+	}
+	if len(tx.Signature) == 0 {
+		tx.Signature = []string{}
+	}
+
+	req, err := sg.newRequest(ctx, http.MethodPost, "/call_contract", tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if opts != nil {
+		vs, err := query.Values(opts)
+		if err != nil {
+			return nil, err
+		}
+		appendQueryValues(req, vs)
+	}
+
+	var resp StarkResp
+	return resp.Result, sg.do(req, &resp)
+}
+
+/*
+	'add_transaction' wrapper for invokation requests
+*/
+func (sg *StarknetGateway) Invoke(ctx context.Context, tx Transaction) (resp AddTxResponse, err error) {
+	tx.EntryPointSelector = BigToHex(GetSelectorFromName(tx.EntryPointSelector))
+	tx.Type = INVOKE
+
+	if len(tx.Calldata) == 0 {
+		tx.Calldata = []string{}
+	}
+	if len(tx.Signature) == 0 {
+		tx.Signature = []string{}
+	}
+
+	req, err := sg.newRequest(ctx, http.MethodPost, "/add_transaction", tx)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, sg.do(req, &resp)
+}
+
+/*
+	'add_transaction' wrapper for compressing and deploying a compiled StarkNet contract
+*/
+func (sg *StarknetGateway) Deploy(ctx context.Context, filePath string, deployRequest DeployRequest) (resp AddTxResponse, err error) {
+	dat, err := os.ReadFile(filePath)
+	if err != nil {
+		return resp, err
+	}
+
+	deployRequest.Type = DEPLOY
+	if len(deployRequest.ConstructorCalldata) == 0 {
+		deployRequest.ConstructorCalldata = []string{}
+	}
+
+	var rawDef RawContractDefinition
+	err = json.Unmarshal(dat, &rawDef)
+	if err != nil {
+		return resp, err
+	}
+
+	deployRequest.ContractDefinition.ABI = rawDef.ABI
+	deployRequest.ContractDefinition.EntryPointsByType = rawDef.EntryPointsByType
+	deployRequest.ContractDefinition.Program, err = CompressCompiledContract(rawDef.Program)
+	if err != nil {
+		return resp, err
+	}
+
+	req, err := sg.newRequest(ctx, http.MethodPost, "/add_transaction", deployRequest)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, sg.do(req, &resp)
+}
+
+func CompressCompiledContract(program map[string]interface{}) (cc string, err error) {
+	pay, err := json.Marshal(program)
+	if err != nil {
+		return cc, err
+	}
+
+	var buf bytes.Buffer
+	zw := gzip.NewWriter(&buf)
+	_, err = zw.Write(pay)
+	if err != nil {
+		return cc, err
+	}
```

Comment:
nit: minimize err scope when possible

```suggestion
	if _, err := zw.Write(pay); err != nil {
		return cc, err
	}
```